week3/homework/src/data/data_dealer.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const fs = require('fs');
+
+const read = function(directory, Unicode) {
+  return new Promise(function(resolve, reject) {
+    fs.readFile(directory, Unicode, function(err, result) {
+      if (err) reject(err);
+      else resolve(result);
+    });
+  });
+};
+
+const write = function(directory, data) {
+  return new Promise(function(resolve, reject) {
+    fs.writeFile(directory, data, err => {
+      if (err) reject(err);
```

Comment:
this either needs an `else` close or a return statement in your `if` condition preventing the `resolve` from being called.

Reply:
I think that I tested it while I have been writing the code. If there is an error then  the code execution will stop after throwing aan error. And for being in the safe side `Else` have to be used.
But Yes, the syntax  of` if statement`  was  confusing for me at that point.